components/family-studies/migrations/src/main/java/org/phenotips/studies/family/migrations/R71500PhenoTips2944DataMigration.java

```diff
@@ -0,0 +1,311 @@
+
+    private static final String PEDIGREE_GENES_FIELD = "genes";
+
+    private static final String PEDIGREE_GENE_FIELD = "gene";
+
+    private static final String PEDIGREE_COLORS_FIELD = "colors";
+
+    private static final String PEDIGREE_SOLVED_GENES_FIELD = "causalGenes";
+
+    private static final String PEDIGREE_CANDIDATE_GENES_FIELD = "candidateGenes";
+
+    private static final String PEDIGREE_REJECTED_GENES_FIELD = "rejectedGenes";
+
+    /** Logging helper object. */
+    @Inject
+    private Logger logger;
+
+    /** Resolves unprefixed document names to the current wiki. */
+    @Inject
+    @Named("current")
+    private DocumentReferenceResolver<String> resolver;
+
+    /** Serializes the class name without the wiki prefix, to be used in the database query. */
+    @Inject
+    @Named("compactwiki")
+    private EntityReferenceSerializer<String> serializer;
+
+    /** Provides access to the all available vocabularies. */
+    @Inject
+    private VocabularyManager vocabularies;
+
+    private Vocabulary hgnc;
+
+    @Override
+    public String getDescription()
+    {
+        return "Translate existing gene data in pedigree editor from gene name to Ensembl gene ID";
+    }
+
+    @Override
+    public XWikiDBVersion getVersion()
+    {
+        return new XWikiDBVersion(71500);
+    }
+
+    @Override
+    public void hibernateMigrate() throws DataMigrationException, XWikiException
+    {
+        getStore().executeWrite(getXWikiContext(), this);
+    }
+
+    @Override
+    public Object doInHibernate(Session session) throws HibernateException, XWikiException
+    {
+        XWikiContext context = getXWikiContext();
+        XWiki xwiki = context.getWiki();
+        this.hgnc = this.vocabularies.getVocabulary(HGNC);
+
+        // Select all families
+        Query q = session.createQuery("select distinct o.name from BaseObject o where o.className = '"
+            + this.serializer.serialize(PEDIGREE_CLASS_REFERENCE)
+            + "' and o.name <> 'PhenoTips.FamilyTemplate'" + "' and o.name <> 'PhenoTips.PatientTemplate'");
+
+        @SuppressWarnings("unchecked")
+        List<String> docs = q.list();
+
+        this.logger.debug("Found {} family documents", docs.size());
+
+        for (String docName : docs) {
+            XWikiDocument familyXDocument;
+            boolean updated = false;
+
+            try {
+                familyXDocument = xwiki.getDocument(this.resolver.resolve(docName), context);
+                if (familyXDocument == null) {
+                    continue;
+                }
+
+                BaseObject pedigreeXObject = familyXDocument.getXObject(PEDIGREE_CLASS_REFERENCE);
+                if (pedigreeXObject == null) {
+                    continue;
+                }
+
+                this.logger.debug("Updating pedigree for family {}.", docName);
+                updated = this.updatePedigreeGenes(pedigreeXObject, context, docName);
```

Comment:
just use `if (!this.updatePedigreeGenes(pedigreeXObject, context, docName)) { continue }`?